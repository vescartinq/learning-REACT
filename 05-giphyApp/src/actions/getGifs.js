export const getGifs = async (category) => {
  const apiKey = 'HPqG9gDBAf3Yu78aQYSjVQp1XvmOPRCV';

  const url = `https://api.giphy.com/v1/gifs/search?q=${encodeURI(
    category
  )}&limit=10&api_key=${apiKey}`;

  const resp = await fetch(url);
  const { data } = await resp.json();

  const gifs = data.map((img) => {
    return {
      id: img.id,
      title: img.title,
      url: img.images?.downsized_medium.url,
    };
  });

  return gifs;
};
