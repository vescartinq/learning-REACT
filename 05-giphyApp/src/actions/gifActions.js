export const getGifs = async (category) => {
  const apiUrl = 'api.giphy.com/v1/gifs';
  const apiKey = 'HPqG9gDBAf3Yu78aQYSjVQp1XvmOPRCV';
  const limitGifs = 20;

  // encodeURI elimina espacios del input por + para la url
  const url = `https://${apiUrl}/search?q=${encodeURI(
    category
  )}&limit=${limitGifs}&api_key=${apiKey}`;

  const resp = await fetch(url);
  const { data } = await resp.json();

  const gifs = data.map((img) => {
    return {
      id: img.id,
      title: img.title,
      url: img.images?.downsized_medium.url,
    };
  });

  return gifs;
};
