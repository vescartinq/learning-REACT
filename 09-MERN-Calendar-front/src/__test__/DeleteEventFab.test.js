import React from 'react';
import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { eventStartDelete } from '../actions/events';
import { DeleteEventFab } from '../components/ui/DeleteEventFab';

jest.mock('./../actions/events.js', () => ({
  eventStartDelete: jest.fn(),
}));

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {};

let store = mockStore(initState);
store.dispatch = jest.fn();

const wrapper = mount(
  <Provider store={store}>
    <MemoryRouter>
      <DeleteEventFab />
    </MemoryRouter>
  </Provider>
);

describe('DeleteEventFab', () => {
  test('should render <DeleteEventFab/> component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should call eventStartDelete when click the button', () => {
    wrapper.find('button').prop('onClick')();

    expect(eventStartDelete).toHaveBeenCalled();
  });
});
