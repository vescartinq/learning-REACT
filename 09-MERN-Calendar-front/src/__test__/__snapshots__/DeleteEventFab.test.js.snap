// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteEventFab should render <DeleteEventFab/> component 1`] = `
<button
  className="btn btn-danger fab-danger"
  onClick={[Function]}
>
  <i
    className="fas fa-trash"
  />
  <span>
     Borrar evento 
  </span>
</button>
`;
